backends/cuda/cuda-vec.cu

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2017, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-734707. All Rights
+// reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+#include "../include/ceed.h"
+#include <cuda.h>
+
+__global__ void setValueK(CeedScalar * __restrict__ vec, CeedInt size, CeedScalar val) {
+  int idx = threadIdx.x + blockDim.x * blockIdx.x;
+  if (idx >= size)
+    return;
+  vec[idx] = val;
+}
+
+extern "C" int DeviceSetValue(CeedScalar* d_array, CeedInt length, CeedScalar val) {
```

Comment:
Namespacing needed. Check `nm -D lib/libceed.so | grep ' [BDT] ' | grep -v -i ' ceed'` after building to see namespace violations.

Reply:
I'm sorry but I'm not following, there is no namespace in C, so what do you mean? Is it because this function is not static? I thought about this but I didn't know how to solve this, so if you know the solution I'd appreciate it.

Reply:
I added `Ceed` in front of `DeviceSetValue`, is it sufficient?

Reply:
C-style namespaces just prefix the library name. You could call this `CeedCudaDeviceSetValue()`, for example. The `nm` command above will give a list of non-namespaced public symbols.

Reply:
Normally this could be `CEED_INTERN`, but it appears that nvcc does not understand visibility attributes.